### ISSUE TITLE
[bug] : Vertical lines at the curve portion by CreateClahe Opencv

### ISSUE THREAD

**Issue by Omkar-Darekar** [opencv/opencv#21104]
##### System information (version)
- Python => 3.7
- OpenCV => 4.5.3.56
- Operating System / Platform => Windows 64 Bit
- Compiler => Spyder

##### Detailed description

I am doing a project related to vein finding using opencv. I have image of hand and after applying CreateClahe on selected ROI of the hand, it is showing vertical lines at the edges of the ROI. 
For CreateClahe, Clip Limit = 40.0 and Tile Size = (9, 9).

##### Steps to reproduce
This is code snippet to reproduce the problem again - 

//Python code 
```
def crop_roi(image, ROIxywh):
    return image[ROIxywh[1]:ROIxywh[1]+ROIxywh[3], ROIxywh[0]:ROIxywh[0]+ROIxywh[2], :]

frame = OrigImageFrame.copy()

#Cropping
CroppedOriginalImage = crop_roi(frame, ROIxywh) 

#Converting to Gray
CroppedOriginalImage = cv2.cvtColor(CroppedOriginalImage, cv2.COLOR_RGB2GRAY 

#Create CLAHE
CroppedOriginalImage = cv2.createCLAHE(clipLimit=40.0, tileGridSize=(9, 9)).apply(CroppedOriginalImage)
            
#Normalize
norm = np.zeros((CroppedImgWidth, CroppedImgHight))
CroppedOriginalImage = cv2.normalize(CroppedOriginalImage,  norm, 0, 255, cv2.NORM_MINMAX)

frame = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY )
CroppedImgHeight, CroppedImgWidth = CroppedOriginalImage.shape[:2]

frame = cv2.cvtColor(frame, cv2.COLOR_GRAY2RGB )

cv2.imwrite("TempImage.bmp", frame)
```

##### Attachment for better understanding - 
Attaching zip file for better understanding. Zip contains 2 .bmp images of hand on which square box will be visible.
Those square box is nothing but cropped portion of hand and applied CreateClahe on it. 
To the right and left side of cropped part/ROI part, you can see straight vertical lines due to CreateClahe. 
[22-Nov-2021.zip](https://github.com/opencv/opencv/files/7586011/22-Nov-2021.zip)

##### Issue submission checklist

 - [ ] I report the issue, it's not a question
 - [ ] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [ ] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by DumDereDum** [comment]
If I understood you correctly, you asked about the line which I marked as `1`. It is the common result to make the most darcness parts with black color. For detailed algo description check [this](https://docs.opencv.org/4.x/d5/daf/tutorial_py_histogram_equalization.html)
I reproduced this line and marked it as `2`, but there is no this line in the `3`rd circle. Additionaly, you can see this black line in the `4`th circle.
If I misunderstood you, please send a photo with the circled line that you mean
![issue](https://user-images.githubusercontent.com/46279571/143267427-dab70ba8-ffbf-49e7-b1fa-e2e8fb162318.jpg)

**Comment by Omkar-Darekar** [comment]
Hello @DumDereDum Thanks for reply.

Circles you drew on the edges of square is not correct.  

I am attaching bellow images for better understanding - 

![ProccessedImage](https://user-images.githubusercontent.com/32523193/143815499-ea2d0258-3b84-4a05-a2f2-1d971c934b54.jpg)
![1](https://user-images.githubusercontent.com/32523193/143815521-5b3206d4-c8e5-4b02-8ae3-388b4eb01ea3.jpg)
![2](https://user-images.githubusercontent.com/32523193/143815524-d502dde7-3b92-46a0-bc26-3aa5c7d16879.jpg)
![3](https://user-images.githubusercontent.com/32523193/143815525-b983f908-ce16-4cfa-b211-e4f60ec26c14.jpg)


Processed Image is the image for comparison.
Please see in "Image : 1, 2, 3", after applying CreateClahe on the hand i.e. ROI, at the curve side of the hand, it is showing vertical lines. 

If we apply EqualizeHist or Clahe filter from ImageJ tool or Clahe feature from Matlab, it is not showing those vertical lines. 
I only see vertical lines after applying CreateClahe. 

For more testing, I am attaching a zip file of sample objects of hand.
[SHG_INPUT_16-11-2021.zip](https://github.com/opencv/opencv/files/7615822/SHG_INPUT_16-11-2021.zip)

**Comment by Omkar-Darekar** [comment]
> If I understood you correctly, you asked about the line which I marked as `1`. It is the common result to make the most darcness parts with black color. For detailed algo description check [this](https://docs.opencv.org/4.x/d5/daf/tutorial_py_histogram_equalization.html) I reproduced this line and marked it as `2`, but there is no this line in the `3`rd circle. Additionaly, you can see this black line in the `4`th circle. If I misunderstood you, please send a photo with the circled line that you mean ![issue](https://user-images.githubusercontent.com/46279571/143267427-dab70ba8-ffbf-49e7-b1fa-e2e8fb162318.jpg)

**Comment by Omkar-Darekar** [comment]
Hello @DumDereDum 

Do I have any updated regarding this issue?

**Comment by alalek** [comment]
> `CroppedOriginalImage = cv2.normalize(CroppedOriginalImage,  norm, 0, 255, cv2.NORM_MINMAX)`

Why are you blaming on CLAHE after this?

---

Usage questions should go to Users OpenCV Q/A forum: https://forum.opencv.org/

**Comment by Omkar-Darekar** [comment]
Hello @alalek 
I appreciate your quick and polite response. 
As you mentioned, if I remove  `CroppedOriginalImage = cv2.normalize(CroppedOriginalImage, norm, 0, 255, cv2.NORM_MINMAX)`
then also I am getting those vertical lines.

I have attached sample images in zip files in above thread. If you have better solution, much appreciated. As I mentioned earlied, I am not getting those vertical lines in EqualizeHist with or without `CroppedOriginalImage = cv2.normalize(CroppedOriginalImage, norm, 0, 255, cv2.NORM_MINMAX)` . 

Thanks in advance.

**Comment by alalek** [comment]
Some vertical/horizontal lines are CLAHE algorithm unavoidable artifacts (due to nature of tiled processing).

Amplification of this you can see here: https://forum.opencv.org/t/how-do-i-remove-artifacts-created-by-clahe-algorithm/2715

Please refer to this tutorial:
https://docs.opencv.org/4.x/d5/daf/tutorial_py_histogram_equalization.html
Original image is available on "2.4" branch (samples/cpp/tsukuba_l.png)
CLAHE sample C++ code is available in samples/tapi/clahe.cpp (run as `./bin/example_tapi_clahe -i=tsukuba.png`)

---

Usage questions should go to Users OpenCV Q/A forum: https://forum.opencv.org/

---

If you still want to report a bug then you should follow provided issue's template and fill:
- minimal complete reproducer including input data and parameters
- expected result
- actual result
